packages/gatsby-source-shopify/src/type-builders/collection-type.ts

```diff
@@ -4,14 +4,14 @@ export function collectionTypeBuilder(prefix: string): string {
         description: String!
         descriptionHtml: String!
         feedback: ${prefix}ResourceFeedback
-        handle: String!
+        handle: Handle!
         id: ID!
         image: ${prefix}Image
         legacyResourceId: String!
         metafield(namespace: String! key: String!): ${prefix}Metafield
         metafields: [${prefix}Metafield!]! @link(from: "metafields___NODE", by: "id")
         products: [${prefix}Product!]! @link(from: "products___NODE", by: "id")
-        productsCount: Int!
+        productsCount: Count!
```

Comment:
Note here for changes in `type-builders` dir - with changes as they are they are breaking changes and would require major bump for shopify plugin (not necessarily bad thing, but just to make sure that implications are clear). 

There is an option to preserve current shape for existing fields in Gatsby meaning users wouldn't need to change their existing queries like this:

```suggestion
        productsCount: Int! @proxy(from: "productsCount.count")
```

and exposing other fields that were introduced in now object and not scalar would be adding new field and not changing existing one, so it would non-breaking change `feat`

```gql
        productsCountPrecision: String! @proxy(from: "productsCount.precision")
```